nodejs/eks/nodegroup.ts

```diff
@@ -454,31 +454,44 @@ ${customUserData}
         minInstancesInService = 0;
     }
 
-    // Set default Name and Kubernetes tags on the ASG's.
-    let autoScalingGroupTags = pulumi.all([
-        eksCluster.name,
-    ]).apply(([clusterName]) => `- Key: Name
-                            Value: ${clusterName}-worker
-                            PropagateAtLaunch: 'true'
-                          - Key: kubernetes.io/cluster/${clusterName}
-                            Value: 'owned'
-                            PropagateAtLaunch: 'true'`);
-
-    // Merge any cluster tags for the ASG's.
-    if (core.tags) {
-        autoScalingGroupTags = pulumi.concat(autoScalingGroupTags, tagsToAsgTags(core.tags));
-    }
-    // Merge any user supplied tags for the ASG's.
+    // Assemble the ASG tags in the proper order to dedupe tags that are inherited,
+    // by using the right-most (last) k/v (in the chance of key collisions)
+    // during the merge of the InputTags maps (via the spread/splat operator):
+    // - core.tags
+    // - args.cloudFormationTags (AWS CF stack auto-tags the ASG w/ all of the CF tags)
+    // - args.autoScalingGroupTags
+    // - Name & clusterName required for Kubernetes
+    //
+    // Since AWS auto-tags the ASG with the tags it inherits from the
+    // CloudFormation stack, we must first dedupe args.autoScalingGroupTags of
+    // any conflicting tags that the CF stack already has / will set on the ASG.
+    let dedupedAsgTags: InputTags = <InputTags>{};
     if (args.autoScalingGroupTags) {
-        autoScalingGroupTags = pulumi.concat(autoScalingGroupTags, tagsToAsgTags(args.autoScalingGroupTags));
+        dedupedAsgTags = args.autoScalingGroupTags;
+        if (args.cloudFormationTags) {
+            dedupedAsgTags = dedupeInputTags(args.cloudFormationTags, args.autoScalingGroupTags);
+        }
```

Comment:
I still feel like we shouldn't really have to do this.  Can't we instead just document what `cloudFormationTags` and `autoScalingGroupTags ` mean, and that tags should be added to either but not both of these.  

Its very subtle that you can provide a `autoScalingGroupTags` entry and it will just silently get dropped if it conflicted with a `cloudFormationTags`.

Reply:
I've made `autoScalingGroupTags` and `cloudFormationTags` mutually exclusive, added in checks, and error msgs.